pkg/image/fetcher.go

```diff
@@ -123,6 +123,37 @@ func (f *Fetcher) Fetch(ctx context.Context, name string, options FetchOptions)
 	return f.fetchDaemonImage(name)
 }
 
+func (f *Fetcher) CheckReadAccessValidator(repo string, options FetchOptions) bool {
+	if !options.Daemon {
+		return f.checkRemoteReadAccess(repo)
+	}
+	if _, err := f.fetchDaemonImage(repo); err != nil {
```

Comment:
We should probably check for `ErrNotFound` and not assume that every error mean that the image is not available.

Reply:
I added a new validation to check if it is `ErrNotFound` execute the actual logic, otherwise just return `false`, also I added a few unit test to check the behavior